### PR TITLE
[DOCS] Adds Google Vertex AI tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a tutorial on using the Google Vertex AI service with the inference API.

### Preview

[Inference API tutorial](https://elasticsearch_bk_112339.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html)

### CODE REVIEW

File: docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc [hunk 1]
```diff
@@ -41,6 +41,13 @@ You can apply for access to Azure OpenAI by completing the form at https://aka.m
 
 // end::azure-ai-studio[]
 
+// tag::google-vertex-ai[]
+* A https://console.cloud.google.com/[Google Cloud account]
+* A project in Google Cloud
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc:46]: Should we add the info that you have to enable Vertex AI API? Also should we mention the permission you need to have? We could also link to the corresponding pages mentioning the permissions you need, which probably makes it a bit more resilient to change on Google's side, but I don't know how detailed we wanna go here.

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 2]
```diff
@@ -178,6 +178,29 @@ Also, when using this model the recommended similarity measure to use in the
...
+// tag::google-vertex-ai[]
+
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/google_vertex_ai_embeddings <1>
+{
+    "service": "googlevertexai",
+    "service_settings": {
+        "service_account_json": "<service_account_json>", <2>
+        "model_id": "text-embedding-004",
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:190]: Should we also explicitly mention where to find the available models?

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 3]
```diff
@@ -178,6 +178,29 @@ Also, when using this model the recommended similarity measure to use in the
...
+        "location": "europe-west3", <3>
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:191]: Should this also be `<location>` placeholder instead of the concrete value for consistency?

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 4]
```diff
@@ -9,16 +9,20 @@ The instructions in this tutorial shows you how to use the {infer} API workflow
...
 IMPORTANT: For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.
 
-The following examples use Cohere's `embed-english-v3.0` model, the `all-mpnet-base-v2` model from HuggingFace, and OpenAI's `text-embedding-ada-002` second generation embedding model.
+The following examples use the:
+
+* `embed-english-v3.0` model for https://docs.cohere.com/docs/cohere-embed[Cohere]
+* `all-mpnet-base-v2` model from https://huggingface.co/sentence-transformers/all-mpnet-base-v2[HuggingFace]
+* `text-embedding-ada-002` second generation embedding model for OpenAI
+* models available through https://ai.azure.com/explore/models?selectedTask=embeddings[Azure AI Studio] or https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models[Azure OpenAI]
+* `text-embedding-004` model for https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api#generative-ai-get-text-embedding-python_vertex_ai_sdk[Google Vertex AI]
```
Comment (timgrein) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:18]: Should we link to https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api instead of the python vertex ai specific section? The top of the page has a model overview

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc [hunk 5]
```diff
@@ -165,6 +165,32 @@ and the `output_field` that will contain the {infer} results.
 
 // end::azure-ai-studio[]
 
+// tag::google-vertex-ai[]
+
+[source,console]
+--------------------------------------------------
+PUT _ingest/pipeline/google_vertex_ai_embeddings
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc:172]: Should we include pipeline in the name here? It's pretty close to the index name `google-vertex-ai-embeddings` and the same as the inference endpoint id. Maybe that could lead to confusion for people not too familiar with the concepts (inference endpoint vs pipeline vs index)

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 6]
```diff
@@ -202,6 +202,38 @@ the {infer} pipeline configuration in the next step.
...
+        "similarity": "dot_product" <4>
+      },
+      "content": { <5>
+        "type": "text" <6>
+      }
+    }
+  }
+}
+--------------------------------------------------
+<1> The name of the field to contain the generated tokens. It must be referenced in the {infer} pipeline configuration in the next step.
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:226]:
```suggestion
<1> The name of the field to contain the generated embeddings. It must be referenced in the {infer} pipeline configuration in the next step.
```

Should this be `embeddings` instead of `tokens`?

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 7]
```diff
@@ -202,6 +202,38 @@ the {infer} pipeline configuration in the next step.
...
+<2> The field to contain the tokens is a `dense_vector` field.
+<3> The output dimensions of the model. This value may be found on the https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings-api[Google Vertex AI model reference].
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:228]: We could also mention that the inference API tries to infer the output dimension automatically, if it's not specified by the user.

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 8]
```diff
@@ -202,6 +202,38 @@ the {infer} pipeline configuration in the next step.
...
+<4> For Google Vertex AI embeddings, the `dot_product` function should be used to calculate similarity.
+<5> The name of the field from which to create the dense vector representation.
+In this example, the name of the field is `content`. It must be referenced in
+the {infer} pipeline configuration in the next step.
+<6> The field type which is text in this example.
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:233]:
```suggestion
<6> The field type which is `text` in this example.
```

File: docs/reference/tab-widgets/inference-api/infer-api-reindex.asciidoc [hunk 9]
```diff
@@ -155,6 +155,29 @@ might affect the throughput of the reindexing process. If this happens, change
 
 // end::azure-ai-studio[]
 
+// tag::google-vertex-ai[]
```
Comment (timgrein) [inline docs/reference/tab-widgets/inference-api/infer-api-reindex.asciidoc:158]: Not part of this PR, but should we link to the reindex API in general in this step and add maybe one sentence why we're using it (basically simulating ingestion)?

**Review by timgrein — COMMENTED** [review]
Nice tutorial 👏 Left some smaller comments, nothing major

**Comment by szabosteve** [comment]
@elasticsearchmachine test this please

**Review by timgrein — APPROVED** [review]
Thank you for addressing the comments, LGTM 🚢

**Comment by szabosteve** [comment]
@elasticsearchmachine update branch

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 10]
```diff
@@ -74,8 +78,8 @@ Once the upload is complete, you can see an index named `test-data` with 182469
 [[reindexing-data-infer]]
 ==== Ingest the data through the {infer} ingest pipeline
 
-Create the embeddings from the text by reindexing the data through the {infer}
-pipeline that uses the chosen model as the inference model.
+Create the embeddings from the text by reindexing the data through the {infer} pipeline that uses the chosen model as the {infer} model.
+This step - with using the {ref}/docs-reindex.html[reindex API] - simulates data ingestion through a pipeline.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:82]:
```suggestion
This step uses the {ref}/docs-reindex.html[reindex API] to simulate data ingestion through a pipeline.
```

File: docs/reference/tab-widgets/inference-api/infer-api-reindex.asciidoc [hunk 11]
```diff
@@ -155,6 +155,29 @@ might affect the throughput of the reindexing process. If this happens, change
...
+  "dest": {
+    "index": "google-vertex-ai-embeddings",
+    "pipeline": "google_vertex_ai_embeddings"
+  }
+}
+----
+// TEST[skip:TBD]
+<1> The default batch size for reindexing is 1000. Reducing `size` to a smaller
+number makes the update of the reindexing process quicker which enables you to
+follow the progress closely and detect errors early.
```
Comment (leemthompo) [inline docs/reference/tab-widgets/inference-api/infer-api-reindex.asciidoc:177]:
```suggestion
<1> The default batch size for reindexing is 1000. Reducing `size` will make updates to the reindexing process faster. This enables you to
follow the progress closely and detect errors early.
```

File: docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc [hunk 12]
```diff
@@ -402,6 +402,71 @@ query from the `azure-ai-studio-embeddings` index sorted by their proximity to t
...
+    "content"
+  ]
+}
+--------------------------------------------------
+// TEST[skip:TBD]
+
+As a result, you receive the top 10 documents that are closest in meaning to the
+query from the `mistral-embeddings` index sorted by their proximity to the query:
+
+[source,consol-result]
```
Comment (leemthompo) [inline docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc:433]:
```suggestion
[source,console-result]
```

Comment (leemthompo) [reply docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc:433]: this might be intentional?

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 13]
```diff
@@ -178,6 +178,30 @@ Also, when using this model the recommended similarity measure to use in the
...
+    "service_settings": {
+        "service_account_json": "<service_account_json>", <2>
+        "model_id": "text-embedding-004", <3>
+        "location": "<location>", <4>
+        "project_id": "<project_id>" <5>
+    }
+}
+------------------------------------------------------------
+// TEST[skip:TBD]
+<1> The task type is `text_embedding` in the path and the `inference_id` which is the unique identifier of the {infer} endpoint is `google_vertex_ai_embeddings`.
```
Comment (leemthompo) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:197]:
```suggestion
<1> The task type is `text_embedding` per the path. `google_vertex_ai_embeddings` is the unique identifier of the {infer} endpoint (its `inference_id`).
```
- just suggestion to split long sentence
- could also make the example something like `google_vertex_unique_id` so it's abundantly obvious

**Review by leemthompo — COMMENTED** [review]
Just a very quick review from me, looks good, just gave few minor suggestions. :)

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 14]
```diff
@@ -74,8 +78,8 @@ Once the upload is complete, you can see an index named `test-data` with 182469
 [[reindexing-data-infer]]
 ==== Ingest the data through the {infer} ingest pipeline
 
-Create the embeddings from the text by reindexing the data through the {infer}
-pipeline that uses the chosen model as the inference model.
+Create the embeddings from the text by reindexing the data through the {infer} pipeline that uses the chosen model as the {infer} model.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:81]:
```suggestion
Create embeddings from the text by reindexing the data through the {infer} pipeline that uses your chosen model.
```
